src/navigation.js

```diff
@@ -1,12 +1,15 @@
+var util = require("./utilities");
+//let makeResNode = util.ResourceNode.makeResNode;
 
 var engine = {};
 
 engine.children = function(coll){
   return coll.reduce(function(acc, x){
-    if(typeof x === 'object'){
-      for (var prop in x) {
-        if(x.hasOwnProperty(prop)) {
-          var v = x[prop];
+    let d = util.valData(x);
+    if(typeof d === 'object'){
+      for (var prop in d) {
+        if(d.hasOwnProperty(prop)) {
```

Comment:
Any reason why Object.keys was not used here in place of the above two lines?

Reply:
Probably not.  The two approaches are nearly the same, and I think for our purposes would be the same.  (We are not using Symbols.)  The tests pass either way.  I think using Object.keys might be more efficient if there are many other inherited properties.  Changed.